src/brickccform.php

```diff
@@ -79,17 +82,76 @@
                 }
                 logTransaction($gateway['name'], $cardInfo, "Successful");
                 $smartyvalues["success"] = true;
+            } elseif (!empty($responseData['secure'])) {
+                $smartyvalues["formHTML"] = $responseData['secure']['formHTML'];
+
+                $_SESSION['invoiceid'] = $_POST["invoiceid"];
+                $_SESSION['invoiceData'] = $invoiceData;

```

Comment:
No need create too many session key,
Just store data for charge request in 1 session key